src/components/navigation-header/components/dropdown-menu/index.tsx

```diff
@@ -0,0 +1,190 @@
+}
+
+/**
+ * A dropdown menu consisiting of an activator button and a dropdown containing
+ * menu item groups.
+ *
+ * TODO: add more details as more interaction support is added
+ */
+const NavigationHeaderDropdownMenu = ({
+  label,
+  itemGroups,
+}: NavigationHeaderDropdownMenuProps) => {
+  const menuRef = useRef<HTMLDivElement>()
+  const activatorButtonRef = useRef<HTMLButtonElement>()
+  const [isOpen, setIsOpen] = useState(false)
+  const numberOfItemGroups = itemGroups.length
+  const menuId = useMemo(() => `menu-${slugify(label)}`, [label])
+
+  // Handles closing the menu if there is a click outside of it and it is open.
+  useOnClickOutside([menuRef], () => setIsOpen(false), isOpen)
+
+  // Handles closing the menu if focus moves outside of it and it is open.
+  useOnFocusOutside(menuRef, () => setIsOpen(false), isOpen)
+
+  /**
+   * Generates a unique ID for a single dropdown menu item based on the ID of
+   * the group it belongs to.
+   */
+  const getItemId = (groupId: string, itemIndex: number): string => {
+    return `${groupId}-item-${itemIndex}`
+  }
+
+  /**
+   * Generates a unique ID for a group of items based on the main menu ID and
+   * the index of the group.
+   */
+  const getItemGroupId = (groupIndex: number): string => {
+    return `${menuId}-itemGroup-${groupIndex}`
+  }
+
+  /**
+   * Handles click interaction with the activator button. When clicked, if the
+   * menu is:
+   *  - open, then it will be closed
+   *  - closed, then it will be opened
+   */
+  const handleClick = () => {
+    setIsOpen(!isOpen)
+  }
+
+  /**
+   * Handles the behavior that should happen when a key is pressed down.
+   * Currently used by both the activator button and each menu item anchor
+   * element. Currently only handles what happens when the Escape is pressed
+   * because all other keyboard interaction is handled by default interactions
+   * with these elements.
+   *
+   * On Escape:
+   *  - the menu is closed, if it is open
+   *  - the activator button is given focus
+   */
+  const handleKeyDown = (e: KeyboardEvent) => {
+    const { isEscapeKey } = deriveKeyEventState(e)
+    if (isEscapeKey) {
+      setIsOpen(false)
+      activatorButtonRef.current.focus()
+    }
+  }
+
+  /**
+   * Handles the start of a mouse hover interaction with the activator button.
+   * When the mouse pointer hovers over the activator button, the menu will be
+   * opened if it is not already open.
+   */
+  const handleMouseEnter = () => {
+    if (!isOpen) {
+      setIsOpen(true)
+    }
+  }
+
+  /**
+   * Handles the end of a mouse hover interaction with the entire menu. If the
+   * menu is open, and the mouse moves outside the bounds either the activator
+   * button or the dropdown menu list, then the menu will be closed.
+   */
+  const handleMouseLeave = () => {
+    if (isOpen) {
+      setIsOpen(false)
+    }
+  }
+
+  return (
+    <div className={s.root} onMouseLeave={handleMouseLeave} ref={menuRef}>
```

Comment:
Have we considered using a tool like hover intent to determine when to show menu on hover? This makes it so that the hover only shows after the mouse has stopped above the trigger to prevent menus opening unintentionally. 

Hook: https://github.com/natelindev/react-use-hoverintent
Demo: https://codesandbox.io/s/lucid-phoebe-5ut4ey?file=/src/App.tsx

Reply:
I've not seen this before! I forked your sandbox to try it out, and when I increase the `interval` option, I see that it works for the mouseover event but not for the mouseout event. Can you tell if I'm using it correctly? https://codesandbox.io/s/beautiful-kowalevski-u0orb3?file=/src/App.tsx

Reply:
Looks like `timeout` effects the `mouseout` event https://github.com/natelindev/react-use-hoverintent#options

Reply:
Ah, that does what I was looking for!

I have some general hesitancy about using either [`react-use-hoverintent`](https://www.npmjs.com/package/react-use-hoverintent) or [`react-hoverintent`](https://www.npmjs.com/package/react-hoverintent) because it's not clear how well supported and maintained they are. Some additional context: I'm not sure if we should keep these hover interactions, but we can't evaluate that at the moment so they're sticking around for now. With all of that in mind, do you think it's fine to make a ticket for evaluating this package outside of this PR before using it? We may also want to consider using react-aria's `useHover` hook, but I'm [still reading](https://react-spectrum.adobe.com/blog/building-a-button-part-2.html) about that one. 😊

Reply:
Yeah, no problem, sounds good.

Reply:
Ticket added! https://app.asana.com/0/1201987349274776/1202074017129387/f